### PR TITLE
BUG: fix `scipy.optimize.show_options` for unknown methods. and add *_METHODS list for each solver script and fix show_options doc

### LINKED ISSUE #10782: BUG: optimize: methods unknown to `scipy.optimize.show_options`
Some `scipy.optimize.minimize` and `scipy.optimize.linprog` solvers aren't known to `scipy.optimize.show_options`.

#### Reproducing code example:

```
from scipy.optimize import show_options
show_options('minimize', 'trust-ncg')  # OK
show_options('minimize', 'trust-constr')  # not OK
show_options('linprog', 'interior-point')  # OK
show_options('linprog', 'revised simplex')  # not OK
```

#### Error message:

```
Traceback (most recent call last):

  File "<ipython-input-9-2adcf805db05>", line 1, in <module>
    show_options('linprog', 'revised simplex')

  File "C:\Users\mhaberla\AppData\Local\Continuum\anaconda3\lib\site-packages\scipy\optimize\optimize.py", line 3104, in show_options
    raise ValueError("Unknown method %r" % (method,))

ValueError: Unknown method 'revised simplex'
  ...
```

#### Scipy/Numpy/Python version information:
```
1.3.0 1.16.4 sys.version_info(major=3, minor=7, micro=4, releaselevel='final', serial=0)
```

### PR DESCRIPTION
**Pull request by AtsushiSakai**
#### Reference issue
fix #10782 

#### What does this implement/fix?
I fixed `scipy.optimize.show_options` for unknown methods and added a test for it.
I added a global *_METHODS list for each solver script which is proposed in #10782 
I fixed `show_options` docstring because it is missing `scipy.optimize.root_scalar` doc links.

### CODE REVIEW

File: scipy/optimize/optimize.py
Comment (mdhaber) [inline scipy/optimize/optimize.py:3399]: Need to mention this in Parameters:
`Type of optimization solver. One of ‘minimize’, ‘minimize_scalar’, ‘root’, or ‘linprog’.`

File: scipy/optimize/_linprog.py
Comment (mdhaber) [inline scipy/optimize/_linprog.py:34]:
```suggestion
LINPROG_METHODS = ['simplex', 'revised simplex', 'interior-point', 'HiGHS', 'HiGHS-ds', 'HiGHS-ipm']
```

Comment (mdhaber) [reply scipy/optimize/_linprog.py:34]: Can you also add these to show_options?

**Comment by mdhaber** [comment]
> I fixed scipy.optimize.show_options for unknown methods and added a test for it.

What was wrong with it for unknown methods?

**Comment by AtsushiSakai** [comment]
@mdhaber I'm sorry this PR is messed up... I will recreate another PR.

**Comment by mdhaber** [comment]
@AtsushiSakai You can use this one again. All it took was to change the branch this is to be merged into from master to something else and then back again.
image

File: scipy/optimize/tests/test_optimize.py
Comment (mdhaber) [inline scipy/optimize/tests/test_optimize.py:2200]:
```suggestion
            # testing that `show_options` works without error
            show_options(solver, method)
            
```
PEP8 newline at the end of file

Comment (mdhaber) [reply scipy/optimize/tests/test_optimize.py:2200]: Add tests for what happens with unrecognized solver and method.

File: scipy/optimize/_minimize.py
Comment (mdhaber) [inline scipy/optimize/_minimize.py:43]:
```suggestion
MINIMIZE_SCALAR_METHODS = ['brent', 'bounded', 'golden']
```
Not at all important, but this would be more consistent.

**Review by mdhaber — CHANGES_REQUESTED** [review]
Thanks for this. Nice cleanup. 

You wrote:
> I fixed `scipy.optimize.show_options` for unknown methods and added a test for it

But this was already working correctly:
```
In [3]: show_options(solver='linprog', method='ekki')
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-3-d5a93c37d4fb> in <module>
----> 1 show_options(solver='linprog', method='ekki')

~\Desktop\scipy\scipy\optimize\optimize.py in show_options(solver, method, disp)
   3515             methods = dict(doc_routines[solver])
   3516             if method not in methods:
-> 3517                 raise ValueError("Unknown method %r" % (method,))
   3518             name = methods[method]
   3519

ValueError: Unknown method 'ekki'
```
I don't see any changes to this part of the code, and I don't see a test for unknown methods (only valid solvers and methods). It would help to add tests that the `ValueError` is raised when the solver/method is unknown.

**Comment by mdhaber** [comment]
If you make these changes and we get CI green, I'll merge this so it's fixed for 1.6.

File: scipy/optimize/tests/test_optimize.py
Comment (mdhaber) [inline scipy/optimize/tests/test_optimize.py:2210]:
```suggestion
        assert_raises(ValueError, show_options, solver, method)
        
```
PEP8 newline.

**Review by mdhaber — CHANGES_REQUESTED** [review]
Thanks for the changes. Besides this, everything else looks good to me! Current failures seem to be intermittent and unrelated, and I don't think the codecov report is correct for this PR, so after this I'll wait for comments from others then squash merge. In the future @AtsushiSakai please review [here](https://docs.scipy.org/doc/numpy-1.15.1/dev/gitwash/development_workflow.html#writing-the-commit-message) for commit message suggestions. For instance, the commit message might have been:
`MAINT: optimize: update show_options with new solvers/methods and add tests`

**Comment by AtsushiSakai** [comment]
@mdhaber Thank you for reviewing and providing the link. I will review it and follow it from the next PR.

**Comment by mdhaber** [comment]
Triggered rebuild. I'll merge when it's done.

**Comment by mdhaber** [comment]
Thanks @AtsushiSakai!

**Pull request merged**
